### PR TITLE
[ENH] Added `DeepARForecaster` to aeon/toolkit

### LINKED ISSUE #3050: [ENH] Implement the DeepARForecaster in `aeon/forecasting/deep_learning`
### Describe the feature or idea you want to propose

I propose the addition of a new forecaster class, `DeepARForecaster`, under the module `aeon.forecasting.deep_learning`. This forecaster will use the DeepAR architecture for probabilistic time series forecasting, built upon aeon’s `BaseDeepForecaster` and a modular `DeepARNetwork` class.

### Describe your proposed solution

The proposed implementation introduces a **`DeepARForecaster`** class which:

- Inherits from:
  - `BaseDeepForecaster`  
  - `IterativeForecastingMixin`  
  - `DirectForecastingMixin`  

- Uses **`DeepARNetwork`** from `aeon.networks._deepar` to construct the underlying architecture.  

- Supports configurable:
  - LSTM units  
  - Dense layers  
  - Dropout  
  - Other DeepAR-specific settings  

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lucifer4073**
#### Reference Issues/PRs

Fixes #3050 .

#### What does this implement/fix? Explain your changes.
Adds DeepAR forecaster to aeon/forecasting/deep_learning

#### Does your contribution introduce a new dependency? If yes, which one?
No

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you **after** the PR has been merged.
- [ ] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [ ] I've added the estimator/function to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

##### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

File: aeon/forecasting/deep_learning/_deepar.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+    def __init__(
+        self,
+        window,
+        horizon=1,
+        batch_size=32,
+        n_epochs=100,
+        verbose=0,
+        optimizer=None,
+        metrics="accuracy",
+        loss=gaussian_loss,
```
Comment (hadifawaz1999) [inline aeon/forecasting/deep_learning/_deepar.py]: should be a string "gaussian_loss" and if its the case use the gaussian_loss and if "mse" use keras mse

File: aeon/forecasting/deep_learning/_deepar.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+    from tensorflow.keras.utils import register_keras_serializable
+
+    @register_keras_serializable(package="Custom")
+    def gaussian_loss(y_true, y_pred):
+        """
+        Gaussian negative log-likelihood loss for concatenated output format.
+
+        Parameters
+        ----------
+        y_true: Ground truth values of shape (batch_size, n_channels)
```
Comment (hadifawaz1999) [inline aeon/forecasting/deep_learning/_deepar.py]: y_true actually have the shape same as y_pred so should fix this in docstring

**Comment by lucifer4073** [comment]
Hi @hadifawaz1999 ,

I have made the required changes. Please review this.

**Comment by hadifawaz1999** [comment]
Thanks @lucifer4073 !
